website/build_site.sh

```diff
@@ -95,13 +95,17 @@ for content_filepath in $(find content/ -type f -name '*.md'); do
     # 2. convert relative paths to be relative to the website root instead
     # 3. strip the .md file ending to replace it with a trailing slash
     # (to be consistent with the hugo md -> html conversion)
+    # 4. Strip any unnecessary '/./' path components we introduce by doing that.
 
     # Also keep a backup for comparison/debugging purposes.
 
     sed -i.bak -r \
         -e "s|\]\(([./]*[^:#)]+)#mlos-github-tree-view\)|](https://github.com/microsoft/MLOS/tree/main/${parent_path}/\1)|g" \
-        -e "s|\]\(([./]*[^:)]+)\)|](/MLOS/${parent_path}/\1)|g" \
-        -e "s|\]\(([./]*[^:#)]+)\.md(#[^)]+)?\)|](\1/\2)|g" \
+        -e "s|\]\(([./]*[^:#)]+)(#[^)]*)?\)|](/MLOS/${parent_path}/\1\2)|g" \
```

Comment:
Makes sure not to swallow the anchor (if one exists) in the first capture so that we don't replace bare `#anchor-in-the-current-page` links with `/this/pages/parent/path/from/root/#anchor-in-the-current-page` (which no longer work).